nucleus/src/main/java/za/co/absa/subatomic/infrastructure/application/view/jpa/BitbucketRepositoryEmbedded.java

```diff
@@ -0,0 +1,24 @@
+package za.co.absa.subatomic.infrastructure.application.view.jpa;
+
+import javax.persistence.Embeddable;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Embeddable
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class BitbucketRepositoryEmbedded {
+
+    private String bitbucketId;
+
+    private String slug;
+
+    private String name;
+
+    private String repoUrl;
```

Comment:
What is contained in these URL fields? Are we storing the full URL? Is there a particular reason for doing that?

Reply:
Yes, we are storing the URL's as per the Bitbucket API calls.

This is for use later, for example, so that we don't have to lookup the `remoteUrl` when configuring the [Build Config](https://github.com/absa-subatomic/quantum-mechanic/blob/fe97f0091e0f77c607f9b55d745d81361646fa00/src/gluon/packages/ApplicationCreated.ts#L173). Also, this will be handy for Photon, which can provide links to to the Application's Bitbucket location etc.

![screen shot 2018-02-20 at 10 08 29](https://user-images.githubusercontent.com/1175891/36413274-572fd884-1626-11e8-8b5e-9cfec9ea4374.png)

Reply:
Ok makes sense for standardising easily across our various systems.